src/test/java/com/znsio/sample/e2e/steps/AmazonCartSteps.java

```diff
@@ -0,0 +1,73 @@
+package com.znsio.sample.e2e.steps;
+
+import com.context.SessionContext;
+import com.context.TestExecutionContext;
+import com.znsio.e2e.entities.Platform;
+import com.znsio.e2e.runner.Runner;
+import com.znsio.e2e.tools.Drivers;
+import com.znsio.sample.e2e.businessLayer.amazonCart.ProductDescriptionPageBL;
+import com.znsio.sample.e2e.businessLayer.amazonCart.SearchResultsListsPageBL;
+import com.znsio.sample.e2e.businessLayer.amazonCart.ShppingCartPageBL;
+import com.znsio.sample.e2e.entities.AMAZON_ASSIGNMENT_TEST_CONTEXT;
+import com.znsio.sample.e2e.businessLayer.amazonCart.AmazonHomePageBL;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.log4j.Logger;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+import java.util.Locale;
+
+public class AmazonCartSteps {
+    private static final Logger LOGGER = Logger.getLogger(AmazonCartSteps.class.getName());
+    private final TestExecutionContext context;
+    private final Drivers allDrivers;
+
+    public AmazonCartSteps() {
+        context = SessionContext.getTestExecutionContext(Thread.currentThread()
+                                                               .getId());
+        LOGGER.info("context: " + context.getTestName());
+        allDrivers = (Drivers) context.getTestState(AMAZON_ASSIGNMENT_TEST_CONTEXT.ALL_DRIVERS);
+        LOGGER.info("allDrivers: " + (null == allDrivers));
+    }
+
+    @Given("user creates search for {string} product on Amazon")
+    public void userCreatesSearchForProductOnAmazon(String product) {
+        LOGGER.info(System.out.printf("userOpensAmazonBaseURL - Persona:'%s', on platform: '%s'", AMAZON_ASSIGNMENT_TEST_CONTEXT.ME,
+                Runner.platform));
+        allDrivers.createDriverFor(AMAZON_ASSIGNMENT_TEST_CONTEXT.ME, Runner.platform, context);
+        context.addTestState(AMAZON_ASSIGNMENT_TEST_CONTEXT.ME, AMAZON_ASSIGNMENT_TEST_CONTEXT.ME);
+        new AmazonHomePageBL(AMAZON_ASSIGNMENT_TEST_CONTEXT.ME, Runner.platform).launchAmazonHomePage();
+        new AmazonHomePageBL().createProductSearch(product);
+        new SearchResultsListsPageBL().searchedProductIsVisible(product);
```

Comment:
Try not to call multiple BL into the steps , one step one BL all the business logic should be done in one method and call that method into the step from BL

Reply:
need more insight and understanding on this

Reply:
done